subprojects/groovy-console/src/main/groovy/groovy/console/ui/Console.groovy

```diff
@@ -1329,6 +1356,47 @@ class Console implements CaretListener, HyperlinkListener, ComponentListener, Fo
         }
     }
 
+    @CompileStatic
+    private Object doRun(boolean selected, SourceType st, HistoryRecord record) {
+        def result
+        def src = record.getTextToRun(selected)
+        if (SourceType.JAVA === st) {
+            Optional<String> optionalPrimaryClassName = findPrimaryClassName(src)
+            if (optionalPrimaryClassName.isPresent()) {
+                def js = new JavaShell(Thread.currentThread().contextClassLoader)
+                js.run(optionalPrimaryClassName.get(), src)
+            }
```

Comment:
I wonder if there should be an else here with a message/error to output area saying "No public class found, nothing to run." or similar.

Reply:
Otherwise, change `public class` to `class` and it looks like it ran but with no output. Whereas, it passes parser but hasn't been compiled or run.

Reply:
It's better to check, but Java compiler/javaparser will complain finally ;-)

Reply:
No, it fails silently. Not the most catastrophic behavior but not as intuitive as possible. We can change later.